lib/puppet/type/yumrepo.rb

```diff
@@ -17,18 +17,24 @@
   ensurable
   # Doc string for properties that can be made 'absent'
   ABSENT_DOC="Set this to `absent` to remove it from the file completely."
-  # False can be false/0/no and True can be true/1/yes in yum.
-  YUM_BOOLEAN=/^(True|False|0|1|No|Yes)$/i
-  YUM_BOOLEAN_DOC="Valid values are: False/0/No or True/1/Yes."
+  # False can be false/0/No/no and True can be true/1/Yes/yes in yum.
+  YUM_BOOLEAN=/^(true|false|0|1|Yes|No|yes|no)$/
+  YUM_BOOLEAN_DOC="Valid values are: false/0/No/no or true/1/Yes/yes."
 
   # Common munge logic for YUM_BOOLEAN values. Munges for two requirements:
   # 1) Because of how regex validation works in Puppet::Parameter::Value,
-  # Boolean false and lowercase false will not be considered invalid. However,
-  # if the user specified false (or true), they meant False (or True).
+  # Boolean false and lowercase false will not be considered invalid.
   # 2) In order for parameter removal to work correctly, when absent is passed
   # as a string it needs to be munged back to a symbol.
   munge_yum_bool = Proc.new do |val|
-    val.to_s == 'absent' ? :absent : val.to_s.capitalize
+    case val.to_s
```

Comment:
Looking at this further (and playing with yum repos), I realized that we don't actually need to alter this munging logic at all. The only change that needs to be made is to remove the capital `True`, `False` `No` and `Yes` from `YUM_BOOLEAN_DOC`.

I believe `YUM_BOOLEAN` is fine with the capital words as we're upcase-ing the input. This is really just an error message cleanup, as it turns out.